src/Integration/Notifications/SonarQubeNotifications.cs

```diff
@@ -0,0 +1,145 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using SonarQube.Client.Models;
+using SonarQube.Client.Services;
+using CancellationTokenSource = System.Threading.CancellationTokenSource;
+
+namespace SonarLint.VisualStudio.Integration.Notifications
+{
+    [Export(typeof(ISonarQubeNotifications))]
+    internal class SonarQubeNotifications : ISonarQubeNotifications
+    {
+        private readonly ITimer timer;
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly CancellationTokenSource cancellation = new CancellationTokenSource();
+
+        private DateTimeOffset lastCheckDate;
+        private string projectKey;
+
+        public INotificationIndicatorViewModel Model { get; private set; }
+
+        public NotificationData GetNotificationData() =>
+            new NotificationData
+                {
+                    IsEnabled = Model.AreNotificationsEnabled,
+                    LastNotificationDate = lastCheckDate
+                };
+
+        [ImportingConstructor]
+        [ExcludeFromCodeCoverage] // Do not unit test MEF constructor
+        internal SonarQubeNotifications(IHost host)
+            : this(host.SonarQubeService, new NotificationIndicatorViewModel(),
+                  new TimerWrapper { Interval = 60000 })
+        {
+        }
+
+        internal SonarQubeNotifications(ISonarQubeService sonarQubeService,
+            INotificationIndicatorViewModel model, ITimer timer)
+        {
+            this.timer = timer;
+            this.sonarQubeService = sonarQubeService;
+            Model = model;
+
+            timer.Elapsed += OnTimerElapsed;
+        }
+
+        public async Task StartAsync(string projectKey, NotificationData notificationData)
+        {
+            if (projectKey == null)
+            {
+                throw new ArgumentNullException(nameof(projectKey));
+            }
+
+            this.projectKey = projectKey;
+
+            Model.AreNotificationsEnabled = notificationData?.IsEnabled ?? true;
+
+            var oneDayAgo = DateTimeOffset.Now.AddDays(-1);
+            if (notificationData == null ||
+                notificationData.LastNotificationDate < oneDayAgo)
+            {
+                lastCheckDate = oneDayAgo;
+            }
+            else
+            {
+                lastCheckDate = notificationData.LastNotificationDate;
+            }
+
+            await UpdateEvents();
+            timer.Start();
+        }
+
+        public void Stop()
+        {
+            cancellation.Cancel();
```

Comment:
Just out of curiosity what happens when you call `Cancel` on an already cancelled token?